mahjong/index.js

```diff
@@ -486,13 +503,22 @@ module.exports = (clients) => {
 				return;
 			}
 
-			if (text === 'ドラ表示牌') {
+			if (text === '手牌') {
 				if (state.phase !== 'gaming') {
 					perdon();
 					return;
 				}
 
-				postMessage(state.ドラ表示牌s.map((牌) => 牌ToName(牌)).join(' '));
+				// 12r588m239p467s東白白 のように表記
+				const categorized手牌Array = Array.from(牌Orders, (_, index) => sort(state.手牌).filter((牌) => get牌Type(牌) === 牌Orders[index]));
```

Comment:
[MAY] 処理の内容が分かりづらいので、[lodash.groupBy](https://lodash.com/docs/#groupBy)などのユーティリティ関数を使用することなどを検討してください。

[MAY] sortが配列の要素数だけ実行されています。そこまで重い処理ではないですが、1度で済む処理は1回しか行わないことが理想的です。

[INFO] ちなみに、上を修正するならこう書く必要はないですが、

```suggestion
				const categorized手牌Array = 牌Orders.map((牌Type) => sort(state.手牌).filter((牌) => get牌Type(牌) === 牌Type));
```

と簡潔に書くことができると思います。

Reply:
lodash.groupBy の返り値が連想配列になってしまうこともあり、hakatashiが提示してくれた修正にしました。